src/addresses.rs

```diff
@@ -0,0 +1,120 @@
+use mimir::rubber::Rubber;
+use mimir::Poi;
+use mimirsbrunn::admin_geofinder::AdminGeoFinder;
+use mimirsbrunn::labels::format_addr_name_and_label;
+use mimirsbrunn::labels::format_street_label;
+use mimirsbrunn::utils::find_country_codes;
+use std::ops::Deref;
+use std::sync::Arc;
+
+/// Read the osm address tags and build a mimir address from them
+///
+/// For the moment we read mostly `addr:city` or `addr:country`
+/// if available we also read `addr:postcode`
+///
+/// We also search for the admins that contains the coordinates of the poi
+/// and add them as the address's admins.
+///
+/// For the moment we do not read `addr:city` or `addr:country` as it could
+/// lead to inconsistency with the admins hierarchy
+pub fn build_new_addr(
+    addr_tag: &str,
+    street_tag: &str,
+    poi: &Poi,
+    admins: Vec<Arc<mimir::Admin>>,
+) -> mimir::Address {
+    let postcode = poi
+        .properties
+        .iter()
+        .find(|p| &p.key == "addr:postcode")
+        .map(|p| p.value.clone());
```

Comment:
Is it really necessary to clone here?

Reply:
yes,
```
error[E0507]: cannot move out of `p.value` which is behind a shared reference
  --> src/addresses.rs:21:18
   |
21 |         .map(|p| p.value);
   |                  ^^^^^^^ move occurs because `p.value` has type `std::string::String`, which does not implement the `Copy` trait
```

Reply:
Can't you use `.as_str()`?

Reply:
We actually need to put this value into a `Vec<String>` on the next line.  
I will change `.clone()` to `.to_owned()` to make the intention clearer.